core/src/main/java/com/amplifyframework/core/model/ModelSchemaRegistry.java

```diff
@@ -119,4 +121,29 @@ public static synchronized ModelSchemaRegistry instance() {
     public void clear() {
         this.modelSchemaMap.clear();
     }
+
+    @Override
+    public boolean equals(@Nullable Object thatObject) {
+        if (this == thatObject) {
+            return true;
+        }
+        if (thatObject == null || getClass() != thatObject.getClass()) {
+            return false;
+        }
+        ModelSchemaRegistry registry = (ModelSchemaRegistry) thatObject;
+        return getModelSchemaMap().equals(registry.getModelSchemaMap());
+    }
+
+    @Override
+    public int hashCode() {
+        return ObjectsCompat.hash(getModelSchemaMap());
```

Comment:
Only edge case I can see here is if you happen to have two objects with identical schemas:

```
Thing1(id, name)
Thing2(id, name)
```
Would these two evaluate to the same hashCode?

Reply:
nvm...This would be comparing two registries. 🤦